src/types.rs

```diff
@@ -246,14 +243,33 @@ pub struct RecordedEvent {
     /// Representing the user-defined metadata associated with this event.
     pub custom_metadata: Bytes,
 
-    /// Indicates wheter the content is internally marked as JSON.
-    pub is_json: bool,
-
     /// An event position in the $all stream.
     pub position: Position,
 }
 
 impl RecordedEvent {
+    /// Type of this event.
+    pub fn event_type(&self) -> &str {
+        if let Some(tpe) = self.metadata.get("type") {
+            &tpe
+        } else {
+            "<no-event-type-provided>"
+        }
+    }
+
+    /// Indicates whether the content is internally marked as JSON.
+    pub fn is_json(&self) -> bool {
+        if let Some(is_json) = self.metadata.get("is-json") {
```

Comment:
It's not how you detect if you are dealing with a JSON payload. Looked at those lines you deleted above, you'll see how it is done.

Reply:
Do you mean that it is now changed to "content-type/application/json"?  I made a mistake and copied the code from the commands.rs ```convert_persistent_proto_recorded_event```.  Is the json check handled differently here for special reasons or is this a bug?  I have reverted the whole RecordedEvent commit because it is decided that caching ```event_type``` and ```is_json``` is better.

Reply:
@YoEight reverted this